insights/parsers/cryptsetup_luksDump.py

```diff
@@ -0,0 +1,172 @@
+============================================================
+This class provides parsing for the output of cryptsetup luksDump
+<device_name>. Outputs from LUKS1 and LUKS2 are supported.
+"""
+
+from insights import parser, Parser, SkipComponent
+from insights.parsers import ParseException
+from insights.parsr import (Literal, AnyChar, Char, Opt, String, WS,
+                            HangingString, WithIndent, Many, EOF, Lift)
+from insights.specs import Specs
+
+import string
+
+
+class DocParser(object):
+    def __init__(self):
+        value_chars = set(string.printable) - set("\n\r")
+
+        FirstLine = Literal("LUKS header information", value="header") << AnyChar.until(Char("\n")) + Opt(Many(Char("\n")))
+        FirstIndent = Literal("  ")
+        # we need to replace the \t by 8 spaces in the input,
+        # otherwise WithIndent does not work properly
+        # SecondIndent = Literal("\t")
+        SecondIndent = Literal(" " * 8)
+
+        Key = String(value_chars - set(":")) << Char(":") % "Key"
+        Value = WS >> HangingString(value_chars) % "Value"
+
+        MultilineContinuation = Many((Char("\n") + Literal(" " * 15) + SecondIndent) >> String(value_chars))
+        Value1 = WS >> String(value_chars) + Opt(MultilineContinuation).map(lambda x: "".join(x)) << Char("\n")
+        Value1 = Value1.map(lambda x: ("".join(x)).strip())
+
+        ZeroLevelKVPair = Key + Value1
+        FirstLevelKVPair = FirstIndent >> Key + Value1
+        SecondLevelKVPair = SecondIndent >> WithIndent(Key + Value)
+
+        LUKS2SectionName = Key << Char("\n")
+        LUKS2SectionEntry = (FirstLevelKVPair + Many(SecondLevelKVPair).map(dict)).map(self.convert_type)
+        LUKS2Section = LUKS2SectionName + Many(LUKS2SectionEntry).map(dict) << Opt(Many(Char("\n")))
+        LUKS2Body = Many(LUKS2Section, lower=1)
+
+        LUKS1Section = ZeroLevelKVPair + Many(SecondLevelKVPair).map(dict) << Opt(Many(Char("\n")))
+        LUKS1Body = Many(LUKS1Section.map(self.convert_status), lower=1)
+
+        KVBlock = Many(Key + Value1).map(dict)
+        LUKS_Header = (FirstLine + KVBlock) << Opt(Many(Char("\n")))
+
+        # LUKS2Body has to go first, because LUKS1Body consumes also valid LUKS2 bodies
+        self.Top = Lift(self.join_header_and_body) * LUKS_Header * (LUKS2Body | LUKS1Body) << EOF
+
+    def join_header_and_body(self, header, body):
+        return dict([header] + body)
+
+    def convert_type(self, section):
+        section[1]["type"] = section[0][1]
+        return [section[0][0], section[1]]
+
+    def convert_status(self, section):
+        section[2]["status"] = section[1]
+        return [section[0], section[2]]
+
+    def __call__(self, content):
+        try:
+            return self.Top(content)
+        except Exception:
+            raise ParseException("There was an exception when parsing one of the outputs of cryptsetup luksDump commands.")
+
+
+@parser(Specs.cryptsetup_luksDump)
+class LUKS_Dump(Parser):
```

Comment:
I think I understand you keep the "LUKS" capitalized, however, per the [PEP 8](https://peps.python.org/pep-0008/#class-names), it would be better to use the `CapsWords` style for the class names to distinguish with the [Type Variable Names](https://peps.python.org/pep-0008/#type-variable-names). Would you mind changing it per `PEP 8`?